types/sanctuary/index.d.ts

```diff
@@ -48,12 +49,17 @@ type NonNegativeInteger     = number;
 
 interface TypeRep {}
 
+// Please retain type-class hierarchy ordering
+// https://github.com/sanctuary-js/sanctuary-type-classes#type-class-hierarchy
 interface Setoid<A> {}
 interface Ord<A> extends Setoid<A> {}
 interface Semigroupoid<A, B> {}
 interface Category<A> extends Semigroupoid<A, A> {}
 interface Semigroup<A> {}
 interface Monoid<A> extends Semigroup<A> {}
+interface Group<A> extends Monoid<A> {
+  'fantasy-land/invert': () => Group<A>;
+}
```

Comment:
I think we should use `{}` here rather than `{ 'fantasy-land/invert': () => Group<A>; }`. I recall there being a reason we could not use the proper interfaces (built-in types, perhaps).

Reply:
If you look at [how Fluture defines its types](https://github.com/fluture-js/Fluture/blob/master/index.d.ts#L26), you can see that the fantasy-land specification are defined within `{}`. My implementation is possibly a little naive...

> I feel the same rationale should work for S.chain and friends.. If the type classes are better defined then they might work. Similar to how [Fluture define Functor and Monad](https://github.com/fluture-js/Fluture/blob/master/index.d.ts#L26).

If this prevents this PR for being merged then I will happily remove it until the implications are better understood. I guess the real test is to add the entire fantasy-land specification and see if typescript can transpile a demo app.. But unfortunately I do not have spare time to try that out at the moment.

Reply:
It makes sense to me to use `{}` for now, then submit a separate pull request to update all the interfaces. :)